pkg/detectors/alchemy/alchemy.go

```diff
@@ -47,6 +48,10 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
+			client := s.client
```

Comment:
__optional__: You could maybe just check `s.client == nil` and use it on line 59 below, rather than creating a client every time.

Reply:
do you mean mutating the scanner struct?

Reply:
Yep. Or set it as a default somewhere? As it is, every time this attempts to verify it will create a new client. Which is fine, but probably not necessary.